src/AddIns/Uno.UI.MediaPlayer.Skia.X11/SharedMediaPlayerExtension.cs

```diff
@@ -135,10 +129,46 @@ private set
 	// On Win32, EnableMouseInput needs to be false, or else libvlc will capture the pointer and we won't receive
 	// any mouse events. Attempting to do this later below doesn't work for some reason. It needs to be
 	// right after constructing the LibVLCSharp.Shared.MediaPlayer
-	internal LibVLCSharp.Shared.MediaPlayer VlcPlayer { get; } = new LibVLCSharp.Shared.MediaPlayer(_vlc) { EnableMouseInput = false, EnableKeyInput = false };
+	internal LibVLCSharp.Shared.MediaPlayer VlcPlayer { get; }
+
+	public static void PreloadVlc()
+	{
+		Task.Run(() =>
+		{
+			if (Interlocked.CompareExchange(ref _vlcInitialized, 1, 0) == 0)
```

Comment:
Let's create an actual Task that we can away later instead of using a interlocked exchange. I would also tend to think that to avoid having to wait in the ctor for `_vlc`, we can create instance here and drop if afterwards, only keeping the one from the ctor.

Reply:
Constructing an MPE instance is sync so we wouldn't have anywhere to await. I remember contructing multiple vlc instances to be problematic, which is why I went with the atomic swapping approach, but let me see if it's actually fine to create 2 vlc instances